pkg/controller/migration/convert/network.go

```diff
@@ -16,34 +16,46 @@ const (
 )
 
 func handleNetwork(c *components, install *Installation) error {
-	// TODO: be more elegant about creation of this spec field if it already exists
-	if install.Spec.CalicoNetwork == nil {
-		install.Spec.CalicoNetwork = &operatorv1.CalicoNetworkSpec{}
+
+	// FELIX_DEFAULTENDPOINTTOHOSTACTION
+	defaultWepAction, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "FELIX_DEFAULTENDPOINTTOHOSTACTION")
+	if err != nil {
+		return err
 	}
+	if defaultWepAction != nil && strings.ToLower(*defaultWepAction) != "accept" {
+		return ErrIncompatibleCluster{
+			fmt.Sprintf("unexpected FELIX_DEFAULTENDPOINTTOHOSTACTION: '%s'. Only 'accept' is supported.", *defaultWepAction),
+		}
+	}
+
+	if c.calicoCNIConfig != nil {
+		return handleCalicoCNI(c, install)
+	} else {
+		return handleNoCNIConfiguration(c, install)
+	}
```

Comment:
again, I think this belongs up a level where handleNetwork is called.

Reply:
I see these as sub-handlers of handleNetwork. But I guess they could be individual handlers. I'll make the switch.